drivers/hmis/app/models/hmis/hud/processors/base.rb

```diff
@@ -151,32 +151,43 @@ def process_custom_field(field, value)
     }
     # Normalize the input value
     value = normalize_custom_field_value(cded, value)
+
+    # `file` is the only field type that saves an additional record, so do some special-case processing
+    value = process_files(cded, value) if cded.field_type == 'file'
+
     # Infer the field name on the CustomDataElement
     value_field_name = "value_#{cded.field_type}"
 
     # Existing CustomDataElement records for this record with this definition
-    existing_values = existing_custom_data_elements[cded.id] || []
+    existing_cdes = existing_custom_data_elements[cded.id] || []
 
     # If this custom field only allows 1 value and there already is one, update it.
-    if !cded.repeats && existing_values.any?
-      cde_attributes = { id: existing_values.first.id }
+    if !cded.repeats && existing_cdes.any?
+      cde_attributes = { id: existing_cdes.first.id }
       if value.nil?
         cde_attributes[:_destroy] = 1
       else
         cde_attributes[value_field_name] = value
         cde_attributes.merge!(attrs)
       end
-    # If value(s) haven't changed, just update the User and timestamps
-    elsif existing_values.map(&:value) == Array.wrap(value)
-      cde_attributes = existing_values.map { |cde| { id: cde.id, user: @processor.hud_user } }
-    # Else create new custom field value(s), and delete any existing ones.
     else
+      submitted = Array.wrap(value)
+
+      # Helper lambda to check if an existing value was also in the submitted values
+      is_in_submitted = ->(cde) { submitted.include?(cde.value) }
+
+      new_values = submitted - existing_cdes.map(&:value)
+      unchanged = existing_cdes.filter(&is_in_submitted)
+      to_remove = existing_cdes.reject(&is_in_submitted)
+
       cde_attributes = [
+        # If value(s) haven't changed, just update the User and timestamps
```

Comment:
so minor, but since this is complicated code,
```suggestion
        # For value(s) that haven't changed, just update the User and timestamps
```